internal/provisioner/eks_provisioner_cluster.go

```diff
@@ -160,29 +196,63 @@ func (provisioner *EKSProvisioner) CheckClusterCreated(cluster *model.Cluster) (
 	return true, nil
 }
 
+// CreateNodes creates the EKS nodes.
+func (provisioner *EKSProvisioner) CreateNodes(cluster *model.Cluster) error {
+
+	eksMetadata := cluster.ProvisionerMetadataEKS
+	if eksMetadata == nil {
+		return errors.New("error: EKS metadata not set when creating EKS NodeGroup")
+	}
+	eksMetadata.ChangeRequest.WorkerName = "worker-1"
+
+	nodeGroup, err := provisioner.awsClient.EnsureEKSNodeGroup(cluster)
+	if err != nil || nodeGroup == nil {
+		return errors.Wrap(err, "failed to ensure EKS NodeGroup")
+	}
+
+	err = provisioner.clusterUpdateStore.UpdateCluster(cluster)
+	if err != nil {
+		return errors.Wrap(err, "failed to store cluster")
+	}
+
+	return nil
+}
+
 // CheckNodesCreated provisions EKS cluster.
 func (provisioner *EKSProvisioner) CheckNodesCreated(cluster *model.Cluster) (bool, error) {
```

Comment:
nit: if this actually creates the nodes, I would rename this to `EnsureNodesCreated`.

Reply:
`CheckNodesCreated` doesn't create NodeGroup. It checks if NodeGroup is created. `CreateNodes` creates NodeGroup

Reply:
I thought it actually involved some modifications since at the end it calls `UpdateCluster`, but I just noticed that just on the DB, not on the cluster. 🤦